main.cpp

```diff
@@ -4,9 +4,7 @@ int main()
 {
     RAClog::FileLogHandler *log =  RAClog::FileLogHandler::CreateInstance();
```

Comment:
Should use a `std::shared_ptr` or `std::unique_ptr` here, thus you could specify the destructor of `RAClog::FileLogHandler` on the same, you will avoid to ever forget to release it !

Reply:
oFileStream should not be static if you want it to be destroyed during the FileLogHandler destruction
If this is because you need it in your log macro, you might want to do add a public method that take a string and that pass it to the file, thus your attribute can remain private

Reply:
Seeing that i declare the FileLoghandler in the main and i destroy it also in the main, and only here, i don't think that a smart pointer will be very usefull, i won't using a smart pointer just for using a smart pointer, i don't see an smart utility here.

Reply:
More than that, seeing that i call the destructor of FileLogHandler at the end of my main function, the destructor of my static object will be destroyed just after it (not in the order i agre, but does'nt matter)

Reply:
Obviously the code is correct as it is, just my 2cents & my taste: I like to see the raw pointer managed by a `std::shared_ptr`, `std::unique_ptr` to be sure that I never forget to release the resources and I know that I do not have to think to how it work in multiple thread environment.